source/connect/tls.txt

```diff
@@ -0,0 +1,241 @@
+by configured Certificate Authority, hostname validation, and expiration.
+
+To overwrite this behavior, it is possible to disable hostname validation, OCSP endpoint revocation checking,
+revocation checking entirely, and allow invalid certificates.
+
+This behavior is controlled using the ``tlsAllowInvalidHostnames``, ``tlsDisableOCSPEndpointCheck``,
+``tlsDisableCertificateRevocationCheck``, and ``tlsAllowInvalidCertificates`` options. By default,
+all are set to ``false``.
+
+It is not recommended to change these defaults as it exposes the client to *Man In The Middle* attacks (when
+``tlsAllowInvalidHostnames`` is set), invalid certificates (when ``tlsAllowInvalidCertificates`` is set), or
+potentially revoked certificates (when ``tlsDisableOCSPEndpointCheck`` or ``tlsDisableCertificateRevocationCheck``
+are set).
+
+Supported Libraries
+-------------------
+
+By default, libmongoc will attempt to find a supported TLS library and enable TLS support. This is controlled
+by the cmake flag ``ENABLE_SSL``, which is set to ``AUTO`` by default. Valid values are:
+
+- ``AUTO`` the default behavior. Link to the system's native TLS library, or attempt to find OpenSSL.
+- ``DARWIN`` link to Secure Transport, the native TLS library on macOS.
+- ``WINDOWS`` link to Secure Channel, the native TLS library on Windows.
+- ``OPENSSL`` link to OpenSSL (libssl). An optional install path may be specified with ``OPENSSL_ROOT``.
+- ``LIBRESSL`` link to LibreSSL's libtls. (LibreSSL's compatible libssl may be linked to by setting ``OPENSSL``).
+- ``OFF`` disable TLS support.
+
+OpenSSL
+~~~~~~~
+
+The MongoDB C Driver uses OpenSSL on Linux and Unix platforms (besides macOS). Industry best practices
+and some regulations require the use of TLS 1.1 or newer, which requires at least OpenSSL 1.0.1. Use the following command
+to check your OpenSSL version:
+
+.. code-block:: bash
+
+   $ openssl version
+
+Ensure your system's OpenSSL is a recent version (at least 1.0.1), or use the following command to install a recent version in a
+non-system path and build against it:
+
+.. code-block:: bash
+
+   cmake -DOPENSSL_ROOT_DIR=/absolute/path/to/openssl
+
+When compiled against OpenSSL, the driver will attempt to load the system default certificate store, as configured by the
+distribution. That can be overridden by setting the ``tlsCAFile`` URI option or with the fields ``ca_file`` and ``ca_dir``
+in the `mongoc_ssl_opt_t <{+api-libmongoc+}/mongoc_ssl_opt_t.html>`__.
+
+The Online Certificate Status Protocol (OCSP) (see `RFC 6960 <https://tools.ietf.org/html/rfc6960>`__) is fully supported
+when using OpenSSL 1.0.1+ with the following notes:
+
+- When a ``crl_file`` is set with `mongoc_ssl_opt_t <{+api-libmongoc+}/mongoc_ssl_opt_t.html>`__ and the ``crl_file`` revokes
+  the server's certificate, the certificate is considered revoked (even if the certificate has a valid stapled OCSP response).
```

Comment:
S: S: I'd introduce that external link and avoid having information in parentheses. Also, since there's only one note in the list, you can just combine the info into one paragraph:

```suggestion
The Online Certificate Status Protocol (OCSP) is fully supported
when using OpenSSL 1.0.1+. However, when a ``crl_file`` is set with `mongoc_ssl_opt_t <{+api-libmongoc+}/mongoc_ssl_opt_t.html>`__ and the ``crl_file`` revokes
  the server's certificate, the certificate is considered revoked, even if the certificate has a valid stapled OCSP response.
  
 .. tip::
 
   For more information about OCSP, see `RFC 6960 <https://tools.ietf.org/html/rfc6960>`__.
```